### PR TITLE
feat(schema-form): add file widget

### PR DESCRIPTION
**Pull request by ricardogoncalves89**
image
image

### CODE REVIEW

File: packages/react-ui-components/src/components/SchemaForm/Templates/ObjectFieldTemplate/ObjectFieldTemplate.tsx [hunk 1]
```diff
@@ -10,7 +10,6 @@ import { DEFAULT_INPUT_CONFIG, DEFAULT_INPUT_INLINE_CONFIG } from './config';
 const ObjectFieldTemplate = <T, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({
 	description,
 	title,
-	required,
```
Comment (djcouto) [inline packages/react-ui-components/src/components/SchemaForm/Templates/ObjectFieldTemplate/ObjectFieldTemplate.tsx:13]: Why did you remove this?

File: packages/react-ui-components/src/components/SchemaForm/Widgets/FileWidget/FileWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+}
+
+type FileInfoType = {
+	dataURL?: string | null;
+	name: string;
+	size: number;
+	type: string;
+};
+
+function processFile(file: File): Promise<FileInfoType> {
```
Comment (djcouto) [inline packages/react-ui-components/src/components/SchemaForm/Widgets/FileWidget/FileWidget.tsx:24]: Move these methods to a utils file 🙏

**Review by djcouto — APPROVED** [review]

File: packages/react-ui-components/src/components/SchemaForm/Widgets/FileWidget/FileWidget.tsx (hunk 2 above)
Comment (miguelpinto98) [reply packages/react-ui-components/src/components/SchemaForm/Widgets/FileWidget/FileWidget.tsx:24]: and define the types in a separated file

File: packages/react-ui-components/src/components/SchemaForm/Widgets/FileWidget/FileWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+}
+
+/**
+ *  The `FileWidget` is a widget for rendering file upload fields.
+ *  It is typically used with a string property with data-url format.
+ */
+function FileWidget<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>(props: WidgetProps<T, S, F>) {
+	const { disabled, readonly, required, multiple, onChange, value, options, uiSchema, schema, label, name, rawErrors = [] } = props;
+
+	const [filesInfo, setFilesInfo] = useState<FileInfoType[]>(Array.isArray(value) ? extractFileInfo(value) : extractFileInfo([value]));
```
Comment (miguelpinto98) [inline packages/react-ui-components/src/components/SchemaForm/Widgets/FileWidget/FileWidget.tsx:134]: should we put under `extractFileInfo` the validation of the value is an array?

File: packages/react-ui-components/src/components/SchemaForm/Widgets/FileWidget/FileWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+					</div>
+				</div>
+			)}
+		</div>
+	);
+}
+
+function extractFileInfo(dataURLs: string[]): FileInfoType[] {
+	return dataURLs
+		.filter(dataURL => dataURL)
```
Comment (miguelpinto98) [inline packages/react-ui-components/src/components/SchemaForm/Widgets/FileWidget/FileWidget.tsx:115]: can we have undefined dataUrls here?

**Review by miguelpinto98 — APPROVED** [review]
